docs/external-node.md

```diff
@@ -0,0 +1,532 @@
+  - [Prerequisites on Kubernetes cluster](#prerequisites-on-kubernetes-cluster)
+  - [Installation on Linux VM](#installation-on-linux-vm)
+    - [Prerequisites](#prerequisites-1)
+    - [Installation](#installation)
+  - [Installation on Windows VM](#installation-on-windows-vm)
+    - [Prerequisites](#prerequisites-2)
+    - [Installation](#installation-1)
+- [Limitations](#limitations)
+<!-- /toc -->
+
+# What is ExternalNode?
+
+`ExternalNode` is a CRD API that enables Antrea to manage the network connectivity
+and security on a Non Kubernetes Node (like a virtual machine or a bare-metal
+server). It supports specifying which network interfaces on the external Node
+are expected to be guarded with Antrea NetworkPolicy rules. The virtual machine
+or bare-metal server represented by an `ExternalNode` resource can be either
+Linux or Windows. "external Node" will be used to represent such a virtual
+machine or a bare-metal server in this document.
+
+Antrea NetworkPolicy is applied to an external Node via the `ExternalEntity`
+resource. `antrea-controller` creates an `ExternalEntity` resource for each
+network interface defined in the `ExternalNode` resource.
+
+`antrea-agent` is running on the external Node, and it controls network
+connectivity and security by attaching the network interface(s) to an OVS bridge.
+A new OpenFlow pipeline has been implemented, dedicated to the ExternalNode
+feature.
+
+You may be interested in using this capability for the below scenarios:
+
+- To apply Antrea NetworkPolicy to an external Node.
+- You want the same security configurations on the external Node for all
+  Operating Systems.
+
+This guide demonstrates how to configure `ExternalNode` to achieve the above
+result.
+
+# Prerequisites
+
+`ExternalNode` is introduced in v1.8 as an alpha feature. A feature gate
+`ExternalNode` must be enabled on the `antrea-controller` and `antrea-agent`in
+the configuration. The configuration for `antrea-controller` is modified in the
+`antrea-config` ConfigMap as follows for the feature to work:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: antrea-config-dcfb6k2hkm
+  namespace: kube-system
+data:
+  antrea-controller.conf: |
+    featureGates:
+      ExternalNode: true
+```
+
+Service `antrea` is created for `antrea-controller` to accept connections from
+`antrea-agent`, and the type is `ClusterIP` by default. Since an external Node
+is running outside the Kubernetes cluster, the `ClusterIP` address is not
+accessible. So a `NodePort` or `LoadBalancer` typed Service for `antrea` is
+required.
+
+Since `antrea-agent` is running on an external Node which is not managed by
+Kubernetes, a single configuration file is required to locate on each machine
+where `antrea-agent` is running. 
+
+For the detail configuration options when running `antrea-agent`on an external
+Node, please refer to the [sample configuration file](../build/yamls/externalnode/conf/antrea-agent.conf).
+
```

Comment:
maybe we should add a sentence here like this one:

```
A further [section](#install-antrea-agent-on-vm) will provide detailed steps for running the `antrea-agent` on a VM.
```

@jianjuns what do you think?

Reply:
Added.

Reply:
updated.